LND crashes when calling listchaintxns when there is an unconfirmed TX pending

### Background

When there is an unconfirmed on-chain transaction for LND's wallet, calling `listchaintxns` over gRPC a nil pointer dereference is produced.

### Your environment

* version of `lnd`: 0.5.0-beta commit=3b2c807288b1b7f40d609533c1e96a510ac5fa6d
* which operating system (`uname -a` on *Nix): Linux 4.15.0-34-generic x86_64
* version of `btcd`, `bitcoind`, or other backend: bitcoind 0.16.2

### Steps to reproduce

Send coins to an address in LND's wallet, then call `listchaintxns` immediately.

### Expected behaviour

No crash.

### Actual behaviour

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x45c334]

goroutine 32192 [running]:
main.(*rpcServer).GetTransactions(0xc4201d9650, 0x11b9520, 0xc4216f2ed0, 0x17bb620, 0xc4201d9650, 0xc421f3f720, 0xc42027de30)
        /go/src/github.com/lightningnetwork/lnd/rpcserver.go:3021 +0x298
github.com/lightningnetwork/lnd/lnrpc._Lightning_GetTransactions_Handler.func1(0x11b9520, 0xc4216f2ed0, 0xe80360, 0x17bb620, 0x1, 0x1, 0x0, 0x0)
        /go/src/github.com/lightningnetwork/lnd/lnrpc/rpc.pb.go:6271 +0x89
github.com/lightningnetwork/lnd/macaroons.(*Service).UnaryServerInterceptor.func1(0x11b9520, 0xc4216f2ed0, 0xe80360, 0x17bb620, 0xc421f3f700, 0xc421f3f720, 0xc420744ab0, 0x4128f8, 0x50, 0xf288a0)
        /go/src/github.com/lightningnetwork/lnd/macaroons/service.go:126 +0x156
github.com/lightningnetwork/lnd/lnrpc._Lightning_GetTransactions_Handler(0xf8cf20, 0xc4201d9650, 0x11b9520, 0xc4216f2ed0, 0xc421de7900, 0xc420167900, 0x0, 0x0, 0x6d93f8182, 0xbedf4edf52cbdd69)
        /go/src/github.com/lightningnetwork/lnd/lnrpc/rpc.pb.go:6273 +0x167
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).processUnaryRPC(0xc42006ca00, 0x11beb80, 0xc4209a0000, 0xc420031f00, 0xc4201d9770, 0x17796b0, 0xc422e41140, 0x0, 0x0)
        /go/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:781 +0x9c8
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).handleStream(0xc42006ca00, 0x11beb80, 0xc4209a0000, 0xc420031f00, 0xc422e41140)
        /go/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:981 +0x1318
github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc42234a710, 0xc42006ca00, 0x11beb80, 0xc4209a0000, 0xc420031f00)
        /go/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:551 +0x9f
created by github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc.(*Server).serveStreams.func1
        /go/src/github.com/lightningnetwork/lnd/vendor/google.golang.org/grpc/server.go:549 +0xa1
```

I have a fix ready, this bug report is mostly for documentation.

--- a/rpcserver.go
+++ b/rpcserver.go
@@ -3014,11 +3014,18 @@ func (r *rpcServer) GetTransactions(ctx context.Context,
 			destAddresses = append(destAddresses, destAddress.EncodeAddress())
 		}
 
+		// We also get unconfirmed transactions, so BlockHash can be
+		// nil.
+		blockHash := ""
+		if tx.BlockHash != nil {
+			blockHash = tx.BlockHash.String()
+		}
+
 		txDetails.Transactions[i] = &lnrpc.Transaction{
 			TxHash:           tx.Hash.String(),
 			Amount:           int64(tx.Value),
 			NumConfirmations: tx.NumConfirmations,
-			BlockHash:        tx.BlockHash.String(),
+			BlockHash:        blockHash,
 			BlockHeight:      tx.BlockHeight,
 			TimeStamp:        tx.Timestamp,
 			TotalFees:        tx.TotalFees,

